src/thor/bidirectional_astar.cc

```diff
@@ -240,15 +240,21 @@ inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
 
   // Get cost. Separate out transition cost.
   Cost transition_cost = costing_->TransitionCost(meta.edge, nodeinfo, pred);
-  Cost newcost = pred.cost() + transition_cost +
-                 costing_->EdgeCost(meta.edge, tile, kConstrainedFlowSecondOfDay);
+  Cost newcost =
+      pred.cost() + transition_cost +
+      costing_->EdgeCost(meta.edge, tile,
+                         kConstrainedFlowSecondOfDay + pred.cost().secs + transition_cost.secs);
+
+  // Secondary constraints not met. Don't add to queue.
+  if (!costing_->ConstraintsSatisfied(newcost))
+    return false;
 
   // Check if edge is temporarily labeled and this path has less cost. If
   // less cost the predecessor is updated and the sort cost is decremented
   // by the difference in real cost (A* heuristic doesn't change)
   if (meta.edge_status->set() == EdgeSet::kTemporary) {
     BDEdgeLabel& lab = edgelabels_forward_[meta.edge_status->index()];
-    if (newcost.cost < lab.cost().cost) {
+    if (newcost.cost < lab.cost().cost && costing_->ConstraintsSatisfied(newcost)) {
```

Comment:
that is a good catch, i was going to mention to do this if you didnt already catch it. excellent work!

Reply:
The change on 257 is unnecessary, since I already perform that check 7 lines above. It's a relic of my initial implementation, when I forgot to disallow the edge if the constraint wasn't met, I cleaned most of that up, but missed that one!